app/models/listing/blocked_date.rb

```diff
@@ -0,0 +1,26 @@
+# == Schema Information
+#
+# Table name: listing_blocked_dates
+#
+#  id         :bigint           not null, primary key
+#  listing_id :bigint
+#  blocked_at :date
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+# Indexes
+#
+#  index_listing_blocked_dates_on_listing_id  (listing_id)
+#
+
+class Listing::BlockedDate < ApplicationRecord
+  belongs_to :listing
+
+  scope :in_period, ->(start_on, end_on) do
+    where('blocked_at >= ? AND blocked_at <= ?', start_on, end_on)
```

Comment:
We could add index also on community_id, listing_id and blocked_at, so that queries like this use index.